### PR TITLE
confu_json new version confu_json/0.0.9

### PR DESCRIPTION
**Pull request by werto87**
Specify library name and version:  **confu_json/0.0.9**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

New version and bug fix for https://github.com/conan-io/conan-center-index/pull/12168#issuecomment-1213064143
---

- [ x ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by werto87** [comment]
I added a new check and now the v2 migration linter raises an error.
`recipes/confu_json/all/conanfile.py:2: [E0611(no-name-in-module), ] No name 'ConanInvalidConfiguration' in module 'conans.errors'`
I do not understand why the error gets raised. I think I did not call something different than before.

File: recipes/confu_json/all/conandata.yml [hunk 1]
```diff
@@ -2,3 +2,18 @@ sources:
   0.0.5:
     url: https://github.com/werto87/confu_json/archive/refs/tags/v0.0.5.tar.gz
     sha256: bc506d4b7836a7689b1c6a2d89bb0c3441f774c8f845fef79d85c71099df5bf9
+  0.0.6:
```
Comment (uilianries) [inline recipes/confu_json/all/conandata.yml:5]: please, add only the latest version and what is really relevant to your project. Avoid adding outdated versions.

File: recipes/confu_json/all/conanfile.py [hunk 2]
```diff
@@ -28,39 +25,46 @@ def _minimum_cpp_standard(self):
     @property
     def _minimum_compilers_version(self):
         return {
-            "Visual Studio": "15",
+            "Visual Studio": "17",
```
Comment (uilianries) [inline recipes/confu_json/all/conanfile.py:28]: Why does it change now?

File: recipes/confu_json/all/conanfile.py [hunk 3]
```diff
@@ -28,39 +25,46 @@ def _minimum_cpp_standard(self):
...
             "gcc": "10",
             "clang": "10",
         }
 
-    def validate(self):
+
+       
+    def configure(self):
+        if self.settings.compiler == "Visual Studio" and Version(self.version)<Version("0.0.10"):
```
Comment (uilianries) [inline recipes/confu_json/all/conanfile.py:36]:
```suggestion
        if is_msvc(self) and Version(self.version) < "0.0.10":
```
from conan.tools.microsoft import is_msvc
from conan.tools.scm import Version

File: recipes/confu_json/all/conanfile.py [hunk 4]
```diff
@@ -28,39 +25,46 @@ def _minimum_cpp_standard(self):
...
         if self.settings.compiler == "apple-clang":
             raise ConanInvalidConfiguration(
                 "apple-clang is not supported because of missing concept support")
         if self.settings.compiler.get_safe("cppstd"):
-            tools.check_min_cppstd(self, self._minimum_cpp_standard)
+         tools.build.check_min_cppstd(self, self._minimum_cpp_standard)
+
         min_version = self._minimum_compilers_version.get(
-            str(self.settings.compiler))
+            str(self.settings.compiler))
```
Comment (uilianries) [inline recipes/confu_json/all/conanfile.py:46]:
```suggestion
            str(self.settings.compiler))
```
extra space

File: recipes/confu_json/all/conanfile.py [hunk 5]
```diff
@@ -28,39 +25,46 @@ def _minimum_cpp_standard(self):
...
         else:
-            if tools.Version(self.settings.compiler.version) < min_version:
+            if Version(self.settings.compiler.version) < min_version:
                 raise ConanInvalidConfiguration(
                     "{} requires C++{} support. "
                     "The current compiler {} {} does not support it.".format(
                         self.name, self._minimum_cpp_standard,
                         self.settings.compiler,
                         self.settings.compiler.version))
+        self.options["boost"].header_only = True
```
Comment (uilianries) [inline recipes/confu_json/all/conanfile.py:59]: Why? Is it really required? Otherwise, use the default Boost option.

**Review by uilianries — COMMENTED** [review]
Thank you for your contribution!

File: recipes/confu_json/all/conanfile.py (hunk 2 above)
Comment (werto87) [reply recipes/confu_json/all/conanfile.py:28]: I only recently tried to compile this library with visual c++. For me it only compiles with visual studio 17

File: recipes/confu_json/config.yml [hunk 6]
```diff
@@ -1,4 +1,13 @@
 versions:
   0.0.5:
-    folder: all    
-
+    folder: all
```
Comment (uilianries) [inline recipes/confu_json/config.yml:3]: Need to update config.yml. according conandata.yml

File: recipes/confu_json/all/conanfile.py [hunk 7]
```diff
@@ -28,25 +27,31 @@ def _minimum_cpp_standard(self):
...
+    def configure(self):
+        if is_msvc(self) and Version(self.version) < "0.0.10":
+            raise ConanInvalidConfiguration(
+                "Visual Studio is not supported in versions before confu_json/0.0.10")
         if self.settings.compiler == "apple-clang":
             raise ConanInvalidConfiguration(
                 "apple-clang is not supported because of missing concept support")
         if self.settings.compiler.get_safe("cppstd"):
-            tools.check_min_cppstd(self, self._minimum_cpp_standard)
+         tools.build.check_min_cppstd(self, self._minimum_cpp_standard)
```
Comment (uilianries) [inline recipes/confu_json/all/conanfile.py:45]:
```suggestion
            tools.build.check_min_cppstd(self, self._minimum_cpp_standard)
```

File: recipes/confu_json/all/conanfile.py [hunk 8]
```diff
@@ -1,10 +1,9 @@
...
-from conans.errors import ConanInvalidConfiguration, ConanException
-import os
-from conans.tools import Version
+from conan import ConanFile, tools
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.scm import Version
+from conan.tools.microsoft import is_msvc
+from conan.tools.scm import Version
 
-required_conan_version = ">=1.33.0"
```
Comment (uilianries) [inline recipes/confu_json/all/conanfile.py:7]: Revert and update to 1.50.0

File: recipes/confu_json/all/test_package/conanfile.py [hunk 9]
```diff
@@ -6,6 +6,9 @@ class TestPackageConan(ConanFile):
     settings = "os", "compiler", "build_type", "arch"
     generators = "cmake"
 
+    def requirements(self):
+        self.requires(self.tested_reference_str)
+
```
Comment (uilianries) [inline recipes/confu_json/all/test_package/conanfile.py:11]:
```suggestion
```
Only if your test package is ported to Conan v2. For instance: https://github.com/conan-io/conan-center-index/pull/12458/files#diff-27f9a641a3f432a849fa0cd6294d8b562be49b5dab88b13b922a0d0d72d2fb94

File: recipes/confu_json/all/conanfile.py [hunk 10]
```diff
@@ -1,11 +1,10 @@
...
-from conans.errors import ConanInvalidConfiguration, ConanException
-import os
-from conans.tools import Version
-
-required_conan_version = ">=1.33.0"
+from conan import ConanFile, tools
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.scm import Version
+from conan.tools.microsoft import is_msvc
+from conan.tools.scm import Version
```
Comment (uilianries) [inline recipes/confu_json/all/conanfile.py:5]:
```suggestion
```
duplicated

**Review by uilianries — APPROVED** [review]
LGTM

File: recipes/confu_json/all/conanfile.py [hunk 11]
```diff
@@ -28,25 +26,31 @@ def _minimum_cpp_standard(self):
...
+
+       
+    def configure(self):
+        if is_msvc(self) and Version(self.version) < "0.0.9":
+            raise ConanInvalidConfiguration(
+                "Visual Studio is not supported in versions before confu_json/0.0.9")
         if self.settings.compiler == "apple-clang":
             raise ConanInvalidConfiguration(
-                "apple-clang is not supported because of missing concept support")
+                "apple-clang is not supported. Pull request welcome")
```
Comment (prince-chrismc) [inline recipes/confu_json/all/conanfile.py:42]: From Conan's perceptive it's a toolchain but I don't mind encourage people to contribute

**Pull request merged**